src/go/k8s/pkg/resources/statefulset_update.go

```diff
@@ -283,11 +356,30 @@ func (r *StatefulSetResource) podEviction(ctx context.Context, pod, artificialPo
 		return nil
 	}
 
-	if *r.pandaCluster.Spec.Replicas > 1 {
-		r.logger.Info("Put broker into maintenance mode",
+	if *r.pandaCluster.Spec.Replicas == 1 {
+		r.logger.Info("Changes in Pod definition other than activeDeadlineSeconds, configurator and Redpanda container name. Deleting pod",
 			"pod-name", pod.Name,
 			"patch", patchResult.Patch)
-		if err = r.putInMaintenanceMode(ctx, int32(ordinal)); err != nil {
+
+		if err = r.Delete(ctx, pod); err != nil {
+			return fmt.Errorf("unable to remove Redpanda pod: %w", err)
+		}
+	}
+	if managedDecommission {
+		log.Info("managed decommission is set: decommission broker")
+		var id *int32
+		if id, err = r.getBrokerIDForPod(ctx, ordinal); err != nil {
+			return err
+		}
+		r.pandaCluster.SetDecommissionBrokerID(id)
+
+		if err = r.handleDecommissionInProgress(ctx, log); err != nil {
+			return err
+		}
+		log.Info("deleting pod")
+	} else {
+		log.Info("Put broker into maintenance mode", "patch", patchResult.Patch)
+		if err = r.putInMaintenanceMode(ctx, ordinal); err != nil {
```

Comment:
As per different comment maintenance mode can not be called if you have replication set to 1.

Reply:
Ah, right. Was supposed to return from the ==1 check.

Reply:
indeed